pkg/operator/configmetrics/configmetrics.go

```diff
@@ -12,39 +15,45 @@ import (
 // of Kubernetes.
 // TODO: in the future this may move to cluster-config-operator.
 func Register(configInformer configinformers.SharedInformerFactory) {
-	prometheus.MustRegister(&configMetrics{
+	legacyregistry.MustRegister(&configMetrics{
 		infrastructureLister: configInformer.Config().V1().Infrastructures().Lister(),
-		cloudProvider: prometheus.NewGaugeVec(prometheus.GaugeOpts{
-			Name: "cluster_infrastructure_provider",
-			Help: "Reports whether the cluster is configured with an infrastructure provider. type is unset if no cloud provider is recognized or set to the constant used by the Infrastructure config. region is set when the cluster clearly identifies a region within the provider. The value is 1 if a cloud provider is set or 0 if it is unset.",
+		cloudProvider: metrics.NewGaugeVec(&metrics.GaugeOpts{
+			Name:           "cluster_infrastructure_provider",
+			StabilityLevel: metrics.ALPHA,
+			Help:           "Reports whether the cluster is configured with an infrastructure provider. type is unset if no cloud provider is recognized or set to the constant used by the Infrastructure config. region is set when the cluster clearly identifies a region within the provider. The value is 1 if a cloud provider is set or 0 if it is unset.",
 		}, []string{"type", "region"}),
 		featuregateLister: configInformer.Config().V1().FeatureGates().Lister(),
-		featureSet: prometheus.NewGaugeVec(prometheus.GaugeOpts{
-			Name: "cluster_feature_set",
-			Help: "Reports the feature set the cluster is configured to expose. name corresponds to the featureSet field of the cluster. The value is 1 if a cloud provider is supported.",
+		featureSet: metrics.NewGaugeVec(&metrics.GaugeOpts{
+			Name:           "cluster_feature_set",
+			StabilityLevel: metrics.ALPHA,
+			Help:           "Reports the feature set the cluster is configured to expose. name corresponds to the featureSet field of the cluster. The value is 1 if a cloud provider is supported.",
 		}, []string{"name"}),
 	})
 }
 
 // configMetrics implements metrics gathering for this component.
 type configMetrics struct {
 	infrastructureLister configlisters.InfrastructureLister
-	cloudProvider        *prometheus.GaugeVec
+	cloudProvider        *metrics.GaugeVec
 	featuregateLister    configlisters.FeatureGateLister
-	featureSet           *prometheus.GaugeVec
+	featureSet           *metrics.GaugeVec
+}
+
+func (m *configMetrics) Create(version *semver.Version) bool {
+	return true
 }
 
 // Describe reports the metadata for metrics to the prometheus collector.
 func (m *configMetrics) Describe(ch chan<- *prometheus.Desc) {
-	ch <- m.cloudProvider.WithLabelValues("", "").Desc()
-	ch <- m.featureSet.WithLabelValues("").Desc()
+	ch <- m.cloudProvider.WithLabelValues("", "").(metrics.Metric).Desc()
+	ch <- m.featureSet.WithLabelValues("").(metrics.Metric).Desc()
```

Comment:
this type conversion is really needed? How ugly.

Reply:
@sttts actually it seems it is

Reply:
@sttts if we use kubemetrics... i switched it back to prometheus.